src/psyclone/psyir/backend/visitor.py

```diff
@@ -253,9 +253,16 @@ def _visit(self, node):
 
                 # Add preceding comment if available
                 if isinstance(node, CommentableMixin):
-                    if node.preceding_comment and self._COMMENT_PREFIX:
-                        result += (self._nindent + self._COMMENT_PREFIX +
-                                   node.preceding_comment + "\n")
+                    # and is in a location that allows line comments
+                    valid_locations = (Schedule, Container)
+                    if not (node.parent and
```

Comment:
I'm not feeling super sharp - please could you add a comment to explain a bit about what this if is excluding/allowing?

Reply:
It was indeed not very nice code, I have now split it in multiple lines and improved the comment.